ksmt-core/src/main/kotlin/org/ksmt/expr/Array.kt

```diff
@@ -31,6 +64,286 @@ sealed class KArrayStoreBase<A : KArraySortBase<R>, R : KSort>(
             addAll(this@KArrayStoreBase.indices)
             add(value)
         }.uncheckedCast()
+
+    /**
+     * Array store indices cache.
+     *
+     * Each node in array store chain contains a cache for only one index.
+     * Cached index is specified by the [cacheKind] property.
+     * Cached indices are rotated.
```

Comment:
That's not obvious to me. Please, add an example with a three-dimensional array. 

How would this chain look for the following store sequence?

```
a = emptyArray
a = a.store((0, 0, 0), 1)
a = a.store((1, 1, 1), 2)
a = a.store((2, 2, 2), 3)
```

The question about how does this chain look like if each node contain information about only one index instead of three